shapely/tests/test_creation.py

```diff
@@ -245,6 +296,56 @@ def test_linearrings_buffer(dim, order):
     assert_geometries_equal(result3, result1[0])
 
 
+@pytest.mark.parametrize(
+    "coords",
+    [
+        [[0, 2], [1, float("nan")], [1, 3], [0, 2]],
+        [[0, 2], [float("nan"), 0], [1, 3], [0, 2]],
+        [[0, 2, 5], [float("nan"), 2, 5], [1, 3, 5], [0, 2, 5]],
+        [[0, 2, 5], [1, 2, float("nan")], [1, 3, 5], [0, 2, 5]],
+    ],
+)
+def test_linearrings_allow_nan(coords):
+    with ignore_invalid():
+        actual = shapely.linearrings(coords, handle_nans="allow")
+    actual = shapely.get_coordinates(actual, include_z=len(coords[0]) == 3)
+
+    assert_array_equal(actual, coords)
+
+
+@pytest.mark.parametrize(
+    "x,y",
+    [
+        ([0, 1, float("nan"), 2, 0], [3, 4, 5, 5, 3]),
+        ([0, 1, 1, 2, 0], [3, 4, float("nan"), 5, 3]),
+        ([0, 1, 2, 0, float("nan")], [3, 4, 5, 3, 3]),
+        ([float("nan"), 0, 1, 2, 0], [3, 3, 4, 5, 3]),
+    ],
+)
+def test_linearrings_ignore_nan(x, y):
+    actual = shapely.linearrings(x, y, handle_nans="ignore")
+    assert_geometries_equal(actual, LinearRing([(0, 3), (1, 4), (2, 5), (0, 3)]))
+
+
+def test_linearrings_ignore_nan_invalid():
+    with pytest.raises(ValueError):
+        shapely.linearrings([1, 2], [np.nan, 2], handle_nans="ignore")
```

Comment:
```suggestion
        shapely.linearrings([0, float("nan"), 0], [3, 4, 3], handle_nans="ignore")
```

So that it covers the corner case of being long enough when allowing NaNs but not when ignoring them

EDIT: changed the suggestion to use `shapely.linearrings([0, float("nan"), 0], [3, 4, 3], handle_nans="ignore")` instead of `shapely.linearrings([0, 1, float("nan"), 0], [3, 4, 5, 3], handle_nans="ignore")`

Reply:
Actually, running that on this branch, this seems to do something wrong:

```
In [16]: shapely.linearrings([0, 1, float("nan"), 0], [3, 4, 5, 3], handle_nans="ignore")
Out[16]: <LINEARRING (0 3, 1 4, 0 3, 0 3)>
```

But, that seems to be our current behaviour anyway, if you indeed ignore the NaN in the above:

```
In [17]: shapely.linearrings([0, 1, 0], [3, 4, 3])
Out[17]: <LINEARRING (0 3, 1 4, 0 3, 0 3)>
```

So even if first and last coordinates are already equal, we do "close" the ring once more if it has a length of 3? (i.e. if it has 3 coordinates, we just always assume the ring has to be closed)

Reply:
This is backwards compatibility. It is there in master, and it was put there to remain compatible with shapely 1.x

Although it is better to just check if the coordiantes are equal, I don't like to be breaking for a shapely 2.1 release.

https://github.com/shapely/shapely/blob/cde4016dfb8d232d548087fe8338ec42a2174e20/src/ufuncs.c#L2596